pycloudlib/errors.py

```diff
@@ -156,3 +156,16 @@ def __init__(self, tag: str, rules_failed: List[str]):
     def __str__(self) -> str:
         """Return string representation of the error."""
         return f"Tag '{self.tag}' failed the following rules: {', '.join(self.rules_failed)}"
+
```

Comment:
Are these exceptions Azure-specific? If so, I think they should live under `pycloudlib/azure/errors.py` and derive from another exception.